### PR TITLE
Ensure non empty parents and self as first item in parents

### PR DESCRIPTION
**Pull request by Fraggle**
## Description

Add self in confluence immediately at first upsert.

## Risk

I don't see any side-effect but pay extra attention.

## Deploy Plan

Deploy `front` and `connector`.

### CODE REVIEW

File: front/lib/api/data_sources.ts [hunk 1]
```diff
@@ -363,13 +363,24 @@ export async function upsertDocument({
...
+  const documentId = name;
+  const documentParents = parents || [];
+
+  // Ensure that the documentId is included in the parents as the first item.
+  // remove it if it's already present and add it as the first item.
+  const indexOfDocumentId = documentParents.indexOf(documentId);
+  if (indexOfDocumentId !== -1) {
+    documentParents.splice(indexOfDocumentId, 1);
+  }
+  documentParents.unshift(documentId);
```
Comment (spolu) [inline front/lib/api/data_sources.ts:375]: This one is not guaranteed ImHo

But as a starter we could add a log with assert if we find parents that don't have the documentId in them?

File: front/lib/api/data_sources.ts [hunk 2]
```diff
@@ -363,13 +363,24 @@ export async function upsertDocument({
   // Data source operations are performed with our credentials.
   const credentials = dustManagedCredentials();
 
+  const documentId = name;
+  const documentParents = parents || [];
+
+  // Ensure that the documentId is included in the parents as the first item.
+  // remove it if it's already present and add it as the first item.
+  const indexOfDocumentId = documentParents.indexOf(documentId);
```
[the remark this answers is not in the document]
Comment (flvndvd) [reply front/lib/api/data_sources.ts:371]: Ok seen IRL, this is only used in Front.

**Review by flvndvd — APPROVED** [review]

File: front/lib/api/data_sources.ts (hunk 1 above)
Comment (Fraggle) [reply front/lib/api/data_sources.ts:375]: I'll drop it for now.

Comment (Fraggle) [reply front/lib/api/data_sources.ts:375]: This is only called in front from the private api when people upsert documents manually, are you sure ?

Comment (spolu) [reply front/lib/api/data_sources.ts:375]: Sorry got it. Misread it. I thought it was a conenctors generic thing. 

That being said it's not limited to folders is it? All connection upserts will run through that code no?

Comment (Fraggle) [reply front/lib/api/data_sources.ts:375]: Hum, beyond folder, outside of connectors, what can upsert ? (right now I only see folder usage, soon attachments uploads but it's also folders).

What do you have in mind ?

Comment (spolu) [reply front/lib/api/data_sources.ts:375]: Connectors. They go through this code path right?

Comment (Fraggle) [reply front/lib/api/data_sources.ts:375]: No (but there is a function with the same name.. can be quite confusing :))

Comment (spolu) [reply front/lib/api/data_sources.ts:375]: kk! Let's maybe rename that function while we're here?

Comment (Fraggle) [reply front/lib/api/data_sources.ts:375]: Yup.

**Pull request merged**